bpf: compile test ENABLE_EGRESS_GATEWAY_COMMON

lmb opened the pull request:
Ensure that only defining ENABLE_EGRESS_GATEWAY_COMMON actually works. Also fix the issues that this uncovered.

Fixes: 6cad3dc324 ("bpf: gate egressgw datapath on separate defines")

julianwiedmann commented on bpf/lib/egress_gateway.h:
```diff
@@ -76,6 +78,9 @@ bool egress_gw_request_needs_redirect(struct iphdr *ip4, int ct_status,
 #else
+	(void)ip4;
+	(void)ct_status;
+	(void)tunnel_endpoint;
```
This is to suppress the warnings about unused parameters, right? Then let's use `__maybe_unused`.

lmb committed:
bpf: compile test ENABLE_EGRESS_GATEWAY_COMMON

Ensure that only defining ENABLE_EGRESS_GATEWAY_COMMON actually
works. Also fix the issues that this uncovered.

Fixes: 6cad3dc324 ("bpf: gate egressgw datapath on separate defines")
Signed-off-by: Lorenz Bauer <lmb@isovalent.com>
